### DISCUSSION TITLE
GLB models not loading from my server, loading fine from Sketchfab

### DISCUSSION THREAD

**Discussion by github-user-77** [mozilla/hubs#3222] [Spoke Questions] [answered]
Hi there. When I try to place a GLB model in Spoke from a url on my server, it says "Error loading model". The exact same model loads perfectly from Sketchfab though. Also, images and sounds load perfectly from the same directory on my server, only the GLB models cause loading error from my server. Any assistance in figuring this out most appreciated!

**Comment by robinkwilson** [comment]
A few questions for us to help pinpoint what's going wrong: 

1. When you try to load your models, **are there any errors in your browser's developer tools** > Console tab (look for red error logs) or Network tab (red failed requests)? **Please paste a screenshot or the text in this discussion thread.**
2. Is this on a Hubs Cloud instance or on hubs.mozilla.com?
3. If you download the models onto your desktop, can you drag the file into the browser window, does it load then?

**Comment by github-user-77** [comment]
@robin-k-wilson thanks very much for following up, I appreciate your time. I'll go backwards through your questions. For context, the exact same GLB model when uploaded to Sketchfab loads into Spoke no problems.

3. Dragging the models into browser window from local computer does not load the model, rather it 'downloads' it.
2. This is on hubs.mozilla.com – I'm about to spin up a Hubs Cloud instance on AWS, so let me know if it is better if I wait to see if it is a problem there before trouble shooting any more.
1. Here is the relevant output from Chrome js console (I redacted my server name, I hope that's okay, let me know if you need it):
```
assets-prod.reticulum.io/spoke/assets/js/project-page-ce834b9d8499cfdc286c.js:1 
o: Error loading model "https://myservername-redacted.com/test/cube2.glb":
  Cause:
    Error resolving media "https://myservername-redacted.com/test/cube2.glb":
      Cause:
        Failed to fetch "https://hubs.mozilla.com/api/v1/media":
          Cause:
            Network Error: 500 Unknown Error. Possibly a CORS error.
    at zi.load (https://assets-prod.reticulum.io/spoke/assets/js/project-page-ce834b9d8499cfdc286c.js:1:128875)
o: Error resolving media "https://myservername-redacted.com/test/cube2.glb":
  Cause:
    Failed to fetch "https://hubs.mozilla.com/api/v1/media":
      Cause:
        Network Error: 500 Unknown Error. Possibly a CORS error.
    at https://assets-prod.reticulum.io/spoke/assets/js/entry-b2c52a5f316c70106cd0.js:2:176320
o: Failed to fetch "https://hubs.mozilla.com/api/v1/media":
  Cause:
    Network Error: 500 Unknown Error. Possibly a CORS error.
    at te.fetch (https://assets-prod.reticulum.io/spoke/assets/js/entry-b2c52a5f316c70106cd0.js:2:188728)
    at async https://assets-prod.reticulum.io/spoke/assets/js/entry-b2c52a5f316c70106cd0.js:2:176183
    at async zi.load (https://assets-prod.reticulum.io/spoke/assets/js/project-page-ce834b9d8499cfdc286c.js:1:127672)
Error: Network Error: 500 Unknown Error. Possibly a CORS error.
    at te.fetch (https://assets-prod.reticulum.io/spoke/assets/js/entry-b2c52a5f316c70106cd0.js:2:188491)
    at async https://assets-prod.reticulum.io/spoke/assets/js/entry-b2c52a5f316c70106cd0.js:2:176183
    at async zi.load (https://assets-prod.reticulum.io/spoke/assets/js/project-page-ce834b9d8499cfdc286c.js:1:127672)
```

**Reply by github-user-77** [reply]
Note: "3, 4, 5" should be "3, 2, 1".

**Comment by robinkwilson** [comment, marked as answer]
It's likely a CORS error since your website isn't explicitly approved by hubs.mozilla.com. On your HC deployment, you can approve your website url in the CORS settings.

Alternatively, you can upload the model to Spoke before adding it to a scene so the model gets downloaded through the proxy and then is hosted by hubs.mozilla.com. This is the preferred method.

Changing CORS settings for Hubs Cloud: https://github.com/mozilla/hubs/discussions/2890

**Reply by robinkwilson** [reply]
Marking as answer for now to get it out of our support pipeline, if not resolved click "Unmark as answer".

**Reply by github-user-77** [reply]
Ok, @robin-k-wilson, I'll see how I go with Hubs Cloud. Thanks again for taking the time.
